iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -388,8 +390,11 @@ private void setHmsTableParameters(String newMetadataLocation, Table tbl, TableM
 
     // If needed set the 'storage_handler' property to enable query from Hive
     if (hiveEngineEnabled) {
-      parameters.put(hive_metastoreConstants.META_TABLE_STORAGE,
-          "org.apache.iceberg.mr.hive.HiveIcebergStorageHandler");
+      String storageHandler = parameters.get(hive_metastoreConstants.META_TABLE_STORAGE);
+      // Check if META_TABLE_STORAGE is not present or does not contain ICEBERG_STORAGE_HANDLER
```

Comment:
or is not an instance of ICEBERG_STORAGE_HANDLER

Reply:
if (storageHandler == null || HiveIcebergStorageHandler.class.isAssignableFrom(
    Class.forName(storageHandler)) {
}

Reply:
Updated